raiden/network/proxies/token_network.py

```diff
@@ -0,0 +1,773 @@
+
+        return channel_data.get('state') == CHANNEL_STATE_OPENED
+
+    def channel_is_closed(self, partner: typing.Address):
+        """ Returns true if the channel is in a closed state, false otherwise. """
+        channel_data = self.detail_channel(partner)
+
+        if not isinstance(channel_data['state'], typing.T_ChannelState):
+            raise ValueError('channel state must be of type ChannelState')
+
+        return channel_data.get('state') == CHANNEL_STATE_CLOSED
+
+    def channel_is_settled(self, partner: typing.Address):
+        """ Returns true if the channel is in a closed state, false otherwise. """
+        channel_data = self.detail_channel(partner)
+
+        if not isinstance(channel_data['state'], typing.T_ChannelState):
+            raise ValueError('channel state must be of type ChannelState')
+
+        return channel_data.get('state') == CHANNEL_STATE_SETTLED
+
+    def closing_address(self, partner: typing.Address):
+        """ Returns the address of the closer, if the channel is closed, None
+        otherwise. """
+        participants_data = self.detail_participants(partner)
+
+        if participants_data.get('our_is_closer'):
+            return participants_data.get('our_address')
+        elif participants_data.get('partner_is_closer'):
+            return participants_data.get('partner_address')
+
+        return None
+
+    def can_transfer(self, partner: typing.Address):
+        """ Returns True if the channel is opened and the node has deposit in
+        it.
+
+        Note: Having a deposit does not imply having a balance for off-chain
+        transfers. """
+        closed = self.channel_is_closed(partner)
+
+        if closed is True:
+            return False
+
+        return self._detail_participant(self.node_address, partner)['deposit'] > 0
+
+    def deposit(self, total_deposit, partner: typing.Address):
+        """ Set total token deposit in the channel to total_deposit.
+
+        Raises:
+            ChannelBusyError: If the channel is busy with another operation
+            RuntimeError: If the token address is empty.
+        """
+        if not isinstance(total_deposit, int):
+            raise ValueError('total_deposit needs to be an integral number.')
+
+        token_address = self.token_address()
+
+        token = Token(
+            self.client,
+            token_address,
+            self.poll_timeout,
+        )
+        current_balance = token.balance_of(self.node_address)
+        current_deposit = self._detail_participant(self.node_address, partner)['deposit']
+        amount_to_deposit = total_deposit - current_deposit
+
+        if amount_to_deposit <= 0:
+            raise ValueError('deposit [{}] must be greater than 0.'.format(
+                amount_to_deposit,
+            ))
+
+        if current_balance < amount_to_deposit:
+            raise ValueError('deposit [{}] cant be larger than the available balance [{}].'.format(
```

Comment:
If this is a user facing error string I would also add the token for which the problem occured